kvmagent/kvmagent/plugins/tensorfusion/executor.py

```
@@ -0,0 +1,403 @@
+        pid = worker.pid
+        proc = self._get_proc(pid)
+        pgid = None
+        try:
+            pgid = os.getpgid(pid)
+            os.killpg(pgid, signal.SIGTERM)
+            logger.info('sent SIGTERM to process group %d (worker %s)' % (pgid, worker.device_uuid))
+        except OSError as e:
+            if e.errno == errno.ESRCH:
+                logger.info('worker %s pid %d already exited before destroy' % (worker.device_uuid, pid))
+            else:
+                raise Exception('failed to kill process group for pid %d: %s' % (pid, str(e)))
```

Ruff (0.15.7)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)